lib/practice_exercises.rb

```diff
@@ -1,13 +1,110 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  low = 0
+  high = string.length - 1
+
+  while low <= high do
+    if /\W/ === string[low]
+      low += 1
+      next
+    end
+
+    if /\W/ === string[high]
+      high -= 1
+      next
+    end
+
+    if string[low].downcase == string[high].downcase
+      low += 1
+      high -= 1
+    else
+      return false
+    end
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word.  Otherwise 👍